drivers/adc/adc_ads114s08.c

```diff
@@ -0,0 +1,906 @@
+		break;
+	case ADC_GAIN_64:
+		ADS114_REGISTER_PGA_GAIN_SET(gain, 0b110);
+		break;
+	case ADC_GAIN_128:
+		ADS114_REGISTER_PGA_GAIN_SET(gain, 0b111);
+		break;
+	default:
+		LOG_ERR("gain value %i not supported", channel_cfg->gain);
+		return -EINVAL;
+	}
+
+	if (channel_cfg->gain != ADC_GAIN_1) {
+		/* enable gain */
+		ADS114_REGISTER_PGA_PGA_EN_SET(gain, 0b01);
+	}
+
+	result = ads114_write_register(&config->bus, ADS114_REGISTER_INPMUX, input_mux);
+
+	if (result != 0) {
+		return result;
+	}
+
+	result = ads114_write_register(&config->bus, ADS114_REGISTER_REF, reference_control);
+
+	if (result != 0) {
+		return result;
+	}
+
+	result = ads114_write_register(&config->bus, ADS114_REGISTER_DATARATE, data_rate);
+
+	if (result != 0) {
+		return result;
+	}
+
+	result = ads114_write_register(&config->bus, ADS114_REGISTER_PGA, gain);
+
+	if (result != 0) {
+		return result;
+	}
+
+	return 0;
+}
+
+static int ads114_validate_buffer_size(const struct adc_sequence *sequence)
+{
+	size_t needed = sizeof(int16_t);
+
+	if (sequence->options) {
+		needed *= (1 + sequence->options->extra_samplings);
+	}
+
+	if (sequence->buffer_size < needed) {
+		return -ENOMEM;
+	}
+
+	return 0;
+}
+
+static int ads114_validate_sequence(const struct device *dev, const struct adc_sequence *sequence)
+{
+	if (sequence->resolution != ADS114_RESOLUTION) {
+		LOG_ERR("invalid resolution");
+		return -EINVAL;
+	}
+
+	if (sequence->channels != BIT(0)) {
+		LOG_ERR("invalid channel");
+		return -EINVAL;
+	}
+
+	if (sequence->oversampling) {
+		LOG_ERR("oversampling is not supported");
+		return -EINVAL;
+	}
+
+	return ads114_validate_buffer_size(sequence);
+}
+
+static void adc_context_update_buffer_pointer(struct adc_context *ctx, bool repeat_sampling)
+{
+	struct ads114_data *data = CONTAINER_OF(ctx, struct ads114_data, ctx);
+
+	if (repeat_sampling) {
+		data->buffer = data->buffer_ptr;
+	}
+}
+
+static void adc_context_start_sampling(struct adc_context *ctx)
+{
+	struct ads114_data *data = CONTAINER_OF(ctx, struct ads114_data, ctx);
+
+	data->buffer_ptr = data->buffer;
+	k_sem_give(&data->acquire_signal);
+}
+
+static int ads114_adc_start_read(const struct device *dev, const struct adc_sequence *sequence,
+				 bool wait)
+{
+	int result;
+	struct ads114_data *data = dev->data;
+
+	result = ads114_validate_sequence(dev, sequence);
+
+	if (result != 0) {
+		return result;
+	}
+
+	data->buffer = sequence->buffer;
+
+	adc_context_start_read(&data->ctx, sequence);
+
+	if (wait) {
+		result = adc_context_wait_for_completion(&data->ctx);
+	}
+
+	return result;
+}
+
+static int ads114_send_start_read(const struct device *dev)
+{
+	const struct ads114_config *config = dev->config;
+	int result;
+
+	result = gpio_pin_set_dt(&config->gpio_start_sync, 1);
+
+	if (result != 0) {
+		LOG_ERR("unable to start ADC operation");
+		return result;
+	}
+
+	k_sleep(K_USEC(ADS114_START_SYNC_PULSE_DURATION_IN_US + ADS114_SETUP_TIME_IN_US));
+	gpio_pin_set_dt(&config->gpio_start_sync, 0);
```

Comment:
No support for START command via SPI?

Reply:
Yes, there is one. I haven't implemented it as I have the signal available in my design. Added the possibility to use the driver without the dedicated START/SYNC signal now anyway.